packages/path_provider/path_provider_android/pigeons/messages.dart

```diff
@@ -42,5 +42,5 @@ abstract class PathProviderApi {
   @TaskQueue(type: TaskQueueType.serialBackgroundThread)
   List<String?> getExternalCachePaths();
   @TaskQueue(type: TaskQueueType.serialBackgroundThread)
-  List<String?> getExternalStoragePaths(StorageDirectory directory);
+  List<String?>? getExternalStoragePaths(StorageDirectory directory);
```

Comment:
Why was this changed? We generally do not use nullable collections, as they are error-prone.

Reply:
Because, according to `PathProviderPlatform` class the `getExternalStoragePaths` returns a nullable List<String>, this is one of the cases that I had to test. But the generated class `TestPathProviderApi`'s function `getExternalStoragePaths` returns List<String?> that is not nullable. If this is also not nullable I can't return a null from this function to mock that case.

Reply:
We don't want to make production code worse to artificially create an extra codepath just so that we test it. If it's impossible to actually get a null result, then the solution is to extract a helper that returns a non-nullable list which both the new method and `getExternalStoragePaths` can delegate to, such that `getDownloadsDirectory` doesn't need to have dead code that handles a case that can't actually happen.

Reply:
Got it 👍. I will fix this.

Reply:
@stuartmorgan I have reverted the changes and extracted that function to return a non-nullable list. Please have a look.